odoo/fields.py

```diff
@@ -1139,6 +1177,137 @@ def update_db_related(self, model):
             join_field=SQL.identifier(join_field),
         ))
 
+    ############################################################################
+    #
+    # SQL generation methods
+    #
+
+    def field_to_sql(self, model: BaseModel, alias: str, property_name: str = '', flush: bool = True) -> SQL:
+        """ Return an :class:`SQL` object that represents the value of the given
+        field from the given table alias.
+
+        The query object is necessary for fields that need to add tables to the query.
+
+        When parameter ``flush`` is true, the method adds some metadata in the
+        result to make method :meth:`~odoo.api.Environment.execute_query` flush
+        the field before executing the query.
+        """
+        if not self.store or not self.column_type:
+            raise ValueError(f"Cannot convert {self} to SQL because it is not stored")
+        if property_name:
+            raise ValueError(f"Property '{property_name}' not supported on {self}")
+        # TODO shouldn't we always flush? (the only exception is id)
+        field_to_flush = self if flush else None
+        return SQL.identifier(alias, self.name, to_flush=field_to_flush)
+
+    def condition_to_sql(self, model: BaseModel, alias: str, fname: str, operator: str, value, query: Query, *, _value_to_column=None) -> SQL:
+        """ Return an :class:`SQL` object that represents the domain condition
+        given by the triple ``(fname, operator, value)`` with the given table
+        alias, and in the context of the given query.
+
+        This method should use the model to resolve the SQL and check access
+        of the field.
+        """
+        sql_field = model._field_to_sql(alias, fname, query)
+        if _value_to_column is None:
+            _value_to_column = lambda v: self.convert_to_column(v, model, validate=False)  # noqa: E731
+
+        # -------------------------------------------------
+        # operator: in
+
+        if operator in ('in', 'not in'):
```

Comment:
TODO make a function per type of condition (relational, equalit, inequality, like)